frontend/src/metabase/reference/components/EditableReferenceHeader.jsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Link } from "react-router";
+import cx from "classnames";
+import pure from "recompose/pure";
+
+import S from "./ReferenceHeader.css";
+import L from "metabase/components/List.css";
+import E from "metabase/reference/components/EditButton.css";
+
+import IconBorder from "metabase/components/IconBorder.jsx";
+import Icon from "metabase/components/Icon.jsx";
+import Ellipsified from "metabase/components/Ellipsified.jsx";
+import EditButton from "metabase/reference/components/EditButton.jsx";
+
+
+const EditableReferenceHeader = ({
+    entity = {},
+    table,
+    type,
+    headerIcon,
+    headerLink,
+    name,
+    user,
+    isEditing,
+    hasSingleSchema,
+    hasDisplayName,
+    startEditing,
+    displayNameFormField,
+    nameFormField
+}) =>
+    <div className="wrapper wrapper--trim">
+        <div className={cx("relative", L.header)} style={type === 'segment' ? {marginBottom: 0} : {}}>
+            <div className={L.leftIcons}>
+                { headerIcon &&
+                    <IconBorder
+                        borderWidth="0"
+                        style={{backgroundColor: "#E9F4F8"}}
+                    >
+                        <Icon
+                            className="text-brand"
+                            name={headerIcon}
+                            width={24}
+                            height={24}
+                        />
+                    </IconBorder>
+                }
+            </div>
+            { type === 'table' && !hasSingleSchema && !isEditing &&
+                <div className={S.headerSchema}>{entity.schema}</div>
+            }
+            <div
+                className={S.headerBody}
+                style={isEditing && name === 'Details' ? {alignItems: "flex-start"} : {}}
+            >
+                { isEditing && name === 'Details' ?
+                    hasDisplayName ?
```

Comment:
The main issue imo is not a ternary operator but the fact that something that imo should be two different components (an edit state and the normal read-only state) are conflated. Rather than two very simple components, each detail and list component has  ternary operators that effectively emit one of two components depending on the edit state.